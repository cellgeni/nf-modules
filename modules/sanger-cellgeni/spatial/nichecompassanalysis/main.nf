process SPATIAL_NICHECOMPASSANALYSIS {
    tag "${meta.id}"
    label 'process_high'

    container "quay.io/cellgeni/nichecompass:0.3.0"

    stageInMode 'copy'

    input:
    tuple val(meta), path(nichecompass_model), path(nichecompass_run_config), path(nichecompass_data)

    output:
    tuple val(meta), path("${out_dir}"), emit: nichecompass_dir
    tuple val(meta), path("${out_dir}_analysis.ipynb"), emit: notebook
    path "versions.yml", emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    out_dir = "${prefix}_nichecompass"
    """
    mkdir ${out_dir}
    papermill \\
        "${moduleDir}/resources/usr/bin/nichecompass_analyse_sample_integration.ipynb" \\
        "${out_dir}_analysis.ipynb" \\
        -p run_root "${out_dir}" \\
        -p cfg_path "${nichecompass_run_config}" \\
        -p model_folder_path "${nichecompass_model}" \\
        ${args} \\
        --kernel python3 \\
        --request-save-on-cell-execute \\
        --progress-bar \\
        --log-level INFO \\
        --log-output


  cat <<-END_VERSIONS > versions.yml
  "${task.process}":
      nichecompass: \$(pip show nichecompass | grep Version | sed -e "s/Version: //g")
  END_VERSIONS
  """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    out_dir = "${prefix}_nichecompass"
    """
    mkdir -p "${out_dir}/figures"
    touch "${out_dir}_analysis.ipynb"

  cat <<-END_VERSIONS > versions.yml
  "${task.process}":
      nichecompass: \$(pip show nichecompass | grep Version | sed -e "s/Version: //g")
  END_VERSIONS
  """
}
