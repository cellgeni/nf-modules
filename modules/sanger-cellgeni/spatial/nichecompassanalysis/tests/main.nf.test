nextflow_process {

    name "Test Process SPATIAL_NICHECOMPASSANALYSIS"
    script "../main.nf"
    process "SPATIAL_NICHECOMPASSANALYSIS"

    tag "modules"
    tag "modules_sanger-cellgeni"
    tag "spatial"
    tag "spatial/nichecompassanalysis"

    test("analysis - 3 samples") {

        when {
            process {
                """
                input[0] = [
                    [ id:'test' ], // meta map
                    file(params.modules_testdata_base_path + "nichecompass/analysis/nichecompass_20250828_170240/artifacts/model/", checkIfExists: true),
                    file(params.modules_testdata_base_path + "nichecompass/analysis/nichecompass_20250828_170240/run_config.json", checkIfExists: true),
                    file(params.modules_testdata_base_path + "nichecompass/analysis/nichecompass_20250828_170240/data/", checkIfExists: true),
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("analysis - 3 samples - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [ id:'test' ], // meta map
                    file(params.modules_testdata_base_path + "nichecompass/analysis/nichecompass_20250828_170240/artifacts/model/", checkIfExists: true),
                    file(params.modules_testdata_base_path + "nichecompass/analysis/nichecompass_20250828_170240/run_config.json", checkIfExists: true),
                    file(params.modules_testdata_base_path + "nichecompass/analysis/nichecompass_20250828_170240/data/", checkIfExists: true),
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
