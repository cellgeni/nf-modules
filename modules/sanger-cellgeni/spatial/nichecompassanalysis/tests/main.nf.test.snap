{
    "h5ad - sample_3": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            [
                                [
                                    "batches_latent_physical_space.svg:md5,34da2e9170bf10ad2b2e021f1c93d984",
                                    "cell_types_latent_physical_space.svg:md5,a0c2e3eac407abd235bc906f7a477a0d",
                                    "enriched_gps_heatmap.svg:md5,8f1d94410879d922367d5673bec873e4",
                                    "log_bayes_factor_2.3_cluster_None_vs_rest_top_20-30_enriched_gps_gp_scores_omics_feature_counts_in_sagittal1_feature_space.png:md5,0bb3de9e244870b495c7a4e859068c95",
                                    "log_bayes_factor_2.3_cluster_None_vs_rest_top_20-30_enriched_gps_gp_scores_omics_feature_counts_in_sagittal2_feature_space.png:md5,278b20743d8bff14d5c61ee5b605dd5a",
                                    "log_bayes_factor_2.3_cluster_None_vs_rest_top_20-30_enriched_gps_gp_scores_omics_feature_counts_in_sagittal3_feature_space.png:md5,9522b0026c83fd301c2f1434084d4c14",
                                    "log_bayes_factor_2.3_niche_enriched_gps_summary.csv:md5,9c91cd8d1c09ef7d0f11b75b2b8cadb3",
                                    "res_0.2_niche_composition_batches.svg:md5,ed2bad923c2cb0db65e60a0ce6d11d4a",
                                    "res_0.2_niche_composition_cell_types.svg:md5,2ba4315e693ad6b3ce3823dfe177d56b",
                                    "res_0.2_niches_latent_physical_space.svg:md5,1149810d79b1abaffff8aed5cc1e4c89"
                                ],
                                [
                                    "attr.pkl:md5,0e9602560d2c665f3c383f9a850171a3",
                                    "model_params.pt:md5,c5e78c72684e62675cd1211e1f0fc046",
                                    "trained.h5ad:md5,d87b9fe09a0ee1611cde9f72b0c655c6",
                                    "var_names.csv:md5,d6db17b25f7a7847eac85d7b5a3552bc"
                                ]
                            ],
                            [
                                [
                                    "gencode.vM25.chr_patch_hapl_scaff.annotation.gtf.gz:md5,c77cbf3cad15af10b6174d4b77c8246f",
                                    "human_mouse_gene_orthologs.csv:md5,b01557bcc3647653d0b323e17fe1e93c"
                                ],
                                [
                                    [
                                        "human_metabolite_enzymes.tsv:md5,242269afd818d1ff93f190271065526c",
                                        "human_metabolite_sensors.tsv:md5,5a02dd9124c603d62e4944526f8b544c",
                                        "mouse_metabolite_enzymes.tsv:md5,f2a48c14af3f5174ae78cf390cbfbe0e",
                                        "mouse_metabolite_sensors.tsv:md5,1987eb7777861838ac8b6224732d88b8"
                                    ],
                                    "nichenet_ligand_target_matrix_v2_mouse.csv:md5,0c00fc0008112adaa551dbab0ce9a855",
                                    "nichenet_lr_network_v2_mouse.csv:md5,c42d99ee4976c592f477f58affff1cd1",
                                    "omnipath_lr_network.csv:md5,03487d78e4fe071fa42832be0bbdbf8d"
                                ]
                            ],
                            "run_config.json:md5,dbc58b55de2275ebfbd20f0a36b28386",
                            "timestamp.txt:md5,026972b602ac18dba8c7a3cf777e0bc7",
                            "train.log:md5,274339ccd3c8f5fe388209523e2e0cbf"
                        ]
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "analysis_20250828_170240.ipynb:md5,8fa95bd07065e7a6c328c7c202240b0a"
                    ]
                ],
                "2": [
                    "versions.yml:md5,8a85556b3ab439938f8dc3b6f42b9b7f"
                ],
                "nichecompass_dir": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            [
                                [
                                    "batches_latent_physical_space.svg:md5,34da2e9170bf10ad2b2e021f1c93d984",
                                    "cell_types_latent_physical_space.svg:md5,a0c2e3eac407abd235bc906f7a477a0d",
                                    "enriched_gps_heatmap.svg:md5,8f1d94410879d922367d5673bec873e4",
                                    "log_bayes_factor_2.3_cluster_None_vs_rest_top_20-30_enriched_gps_gp_scores_omics_feature_counts_in_sagittal1_feature_space.png:md5,0bb3de9e244870b495c7a4e859068c95",
                                    "log_bayes_factor_2.3_cluster_None_vs_rest_top_20-30_enriched_gps_gp_scores_omics_feature_counts_in_sagittal2_feature_space.png:md5,278b20743d8bff14d5c61ee5b605dd5a",
                                    "log_bayes_factor_2.3_cluster_None_vs_rest_top_20-30_enriched_gps_gp_scores_omics_feature_counts_in_sagittal3_feature_space.png:md5,9522b0026c83fd301c2f1434084d4c14",
                                    "log_bayes_factor_2.3_niche_enriched_gps_summary.csv:md5,9c91cd8d1c09ef7d0f11b75b2b8cadb3",
                                    "res_0.2_niche_composition_batches.svg:md5,ed2bad923c2cb0db65e60a0ce6d11d4a",
                                    "res_0.2_niche_composition_cell_types.svg:md5,2ba4315e693ad6b3ce3823dfe177d56b",
                                    "res_0.2_niches_latent_physical_space.svg:md5,1149810d79b1abaffff8aed5cc1e4c89"
                                ],
                                [
                                    "attr.pkl:md5,0e9602560d2c665f3c383f9a850171a3",
                                    "model_params.pt:md5,c5e78c72684e62675cd1211e1f0fc046",
                                    "trained.h5ad:md5,d87b9fe09a0ee1611cde9f72b0c655c6",
                                    "var_names.csv:md5,d6db17b25f7a7847eac85d7b5a3552bc"
                                ]
                            ],
                            [
                                [
                                    "gencode.vM25.chr_patch_hapl_scaff.annotation.gtf.gz:md5,c77cbf3cad15af10b6174d4b77c8246f",
                                    "human_mouse_gene_orthologs.csv:md5,b01557bcc3647653d0b323e17fe1e93c"
                                ],
                                [
                                    [
                                        "human_metabolite_enzymes.tsv:md5,242269afd818d1ff93f190271065526c",
                                        "human_metabolite_sensors.tsv:md5,5a02dd9124c603d62e4944526f8b544c",
                                        "mouse_metabolite_enzymes.tsv:md5,f2a48c14af3f5174ae78cf390cbfbe0e",
                                        "mouse_metabolite_sensors.tsv:md5,1987eb7777861838ac8b6224732d88b8"
                                    ],
                                    "nichenet_ligand_target_matrix_v2_mouse.csv:md5,0c00fc0008112adaa551dbab0ce9a855",
                                    "nichenet_lr_network_v2_mouse.csv:md5,c42d99ee4976c592f477f58affff1cd1",
                                    "omnipath_lr_network.csv:md5,03487d78e4fe071fa42832be0bbdbf8d"
                                ]
                            ],
                            "run_config.json:md5,dbc58b55de2275ebfbd20f0a36b28386",
                            "timestamp.txt:md5,026972b602ac18dba8c7a3cf777e0bc7",
                            "train.log:md5,274339ccd3c8f5fe388209523e2e0cbf"
                        ]
                    ]
                ],
                "notebook": [
                    [
                        {
                            "id": "test"
                        },
                        "analysis_20250828_170240.ipynb:md5,8fa95bd07065e7a6c328c7c202240b0a"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,8a85556b3ab439938f8dc3b6f42b9b7f"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-08-29T14:37:46.289822348"
    },
    "analysis - 3 samples": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            "batches_latent_physical_space.svg:md5,e255fa3e0ce41ea9071326b799dd9a9d",
                            "cell_types_latent_physical_space.svg:md5,51276e25baaeae1a93e119a07c08a03a",
                            "enriched_gps_heatmap.svg:md5,6e1b5d7b229402b5e6d291bf39b79f3e",
                            "log_bayes_factor_2.3_cluster_None_vs_rest_top_20-30_enriched_gps_gp_scores_omics_feature_counts_in_sagittal1_feature_space.png:md5,0bb3de9e244870b495c7a4e859068c95",
                            "log_bayes_factor_2.3_cluster_None_vs_rest_top_20-30_enriched_gps_gp_scores_omics_feature_counts_in_sagittal2_feature_space.png:md5,278b20743d8bff14d5c61ee5b605dd5a",
                            "log_bayes_factor_2.3_cluster_None_vs_rest_top_20-30_enriched_gps_gp_scores_omics_feature_counts_in_sagittal3_feature_space.png:md5,9522b0026c83fd301c2f1434084d4c14",
                            "log_bayes_factor_2.3_niche_enriched_gps_summary.csv:md5,9c91cd8d1c09ef7d0f11b75b2b8cadb3",
                            "res_0.2_niche_composition_batches.svg:md5,d4cbb58d87ec194486b16d6678a15acb",
                            "res_0.2_niche_composition_cell_types.svg:md5,8fc8e2f86b68a2a506de2d29dca3b03b",
                            "res_0.2_niches_latent_physical_space.svg:md5,d5769aceb2b58b3c158c36c69fbce608"
                        ]
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "analysis.ipynb:md5,7bf765e5c0e48d04586b25062c3e13a9"
                    ]
                ],
                "2": [
                    "versions.yml:md5,8a85556b3ab439938f8dc3b6f42b9b7f"
                ],
                "nichecompass_figures": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            "batches_latent_physical_space.svg:md5,e255fa3e0ce41ea9071326b799dd9a9d",
                            "cell_types_latent_physical_space.svg:md5,51276e25baaeae1a93e119a07c08a03a",
                            "enriched_gps_heatmap.svg:md5,6e1b5d7b229402b5e6d291bf39b79f3e",
                            "log_bayes_factor_2.3_cluster_None_vs_rest_top_20-30_enriched_gps_gp_scores_omics_feature_counts_in_sagittal1_feature_space.png:md5,0bb3de9e244870b495c7a4e859068c95",
                            "log_bayes_factor_2.3_cluster_None_vs_rest_top_20-30_enriched_gps_gp_scores_omics_feature_counts_in_sagittal2_feature_space.png:md5,278b20743d8bff14d5c61ee5b605dd5a",
                            "log_bayes_factor_2.3_cluster_None_vs_rest_top_20-30_enriched_gps_gp_scores_omics_feature_counts_in_sagittal3_feature_space.png:md5,9522b0026c83fd301c2f1434084d4c14",
                            "log_bayes_factor_2.3_niche_enriched_gps_summary.csv:md5,9c91cd8d1c09ef7d0f11b75b2b8cadb3",
                            "res_0.2_niche_composition_batches.svg:md5,d4cbb58d87ec194486b16d6678a15acb",
                            "res_0.2_niche_composition_cell_types.svg:md5,8fc8e2f86b68a2a506de2d29dca3b03b",
                            "res_0.2_niches_latent_physical_space.svg:md5,d5769aceb2b58b3c158c36c69fbce608"
                        ]
                    ]
                ],
                "notebook": [
                    [
                        {
                            "id": "test"
                        },
                        "analysis.ipynb:md5,7bf765e5c0e48d04586b25062c3e13a9"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,8a85556b3ab439938f8dc3b6f42b9b7f"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-08-30T15:51:58.842646793"
    },
    "mouse - 3 - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        [
                            [
                                [
                                    
                                ],
                                [
                                    "attr.pkl:md5,0e9602560d2c665f3c383f9a850171a3",
                                    "model_params.pt:md5,c5e78c72684e62675cd1211e1f0fc046",
                                    "trained.h5ad:md5,d87b9fe09a0ee1611cde9f72b0c655c6",
                                    "var_names.csv:md5,d6db17b25f7a7847eac85d7b5a3552bc"
                                ]
                            ],
                            [
                                [
                                    "gencode.vM25.chr_patch_hapl_scaff.annotation.gtf.gz:md5,c77cbf3cad15af10b6174d4b77c8246f",
                                    "human_mouse_gene_orthologs.csv:md5,b01557bcc3647653d0b323e17fe1e93c"
                                ],
                                [
                                    [
                                        "human_metabolite_enzymes.tsv:md5,242269afd818d1ff93f190271065526c",
                                        "human_metabolite_sensors.tsv:md5,5a02dd9124c603d62e4944526f8b544c",
                                        "mouse_metabolite_enzymes.tsv:md5,f2a48c14af3f5174ae78cf390cbfbe0e",
                                        "mouse_metabolite_sensors.tsv:md5,1987eb7777861838ac8b6224732d88b8"
                                    ],
                                    "nichenet_ligand_target_matrix_v2_mouse.csv:md5,0c00fc0008112adaa551dbab0ce9a855",
                                    "nichenet_lr_network_v2_mouse.csv:md5,c42d99ee4976c592f477f58affff1cd1",
                                    "omnipath_lr_network.csv:md5,03487d78e4fe071fa42832be0bbdbf8d"
                                ]
                            ],
                            "run_config.json:md5,dbc58b55de2275ebfbd20f0a36b28386",
                            "timestamp.txt:md5,026972b602ac18dba8c7a3cf777e0bc7",
                            "train.log:md5,274339ccd3c8f5fe388209523e2e0cbf"
                        ]
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "analysis_20250828_170240.ipynb:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    "versions.yml:md5,8a85556b3ab439938f8dc3b6f42b9b7f"
                ],
                "nichecompass_dir": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        [
                            [
                                [
                                    
                                ],
                                [
                                    "attr.pkl:md5,0e9602560d2c665f3c383f9a850171a3",
                                    "model_params.pt:md5,c5e78c72684e62675cd1211e1f0fc046",
                                    "trained.h5ad:md5,d87b9fe09a0ee1611cde9f72b0c655c6",
                                    "var_names.csv:md5,d6db17b25f7a7847eac85d7b5a3552bc"
                                ]
                            ],
                            [
                                [
                                    "gencode.vM25.chr_patch_hapl_scaff.annotation.gtf.gz:md5,c77cbf3cad15af10b6174d4b77c8246f",
                                    "human_mouse_gene_orthologs.csv:md5,b01557bcc3647653d0b323e17fe1e93c"
                                ],
                                [
                                    [
                                        "human_metabolite_enzymes.tsv:md5,242269afd818d1ff93f190271065526c",
                                        "human_metabolite_sensors.tsv:md5,5a02dd9124c603d62e4944526f8b544c",
                                        "mouse_metabolite_enzymes.tsv:md5,f2a48c14af3f5174ae78cf390cbfbe0e",
                                        "mouse_metabolite_sensors.tsv:md5,1987eb7777861838ac8b6224732d88b8"
                                    ],
                                    "nichenet_ligand_target_matrix_v2_mouse.csv:md5,0c00fc0008112adaa551dbab0ce9a855",
                                    "nichenet_lr_network_v2_mouse.csv:md5,c42d99ee4976c592f477f58affff1cd1",
                                    "omnipath_lr_network.csv:md5,03487d78e4fe071fa42832be0bbdbf8d"
                                ]
                            ],
                            "run_config.json:md5,dbc58b55de2275ebfbd20f0a36b28386",
                            "timestamp.txt:md5,026972b602ac18dba8c7a3cf777e0bc7",
                            "train.log:md5,274339ccd3c8f5fe388209523e2e0cbf"
                        ]
                    ]
                ],
                "notebook": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "analysis_20250828_170240.ipynb:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,8a85556b3ab439938f8dc3b6f42b9b7f"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-08-29T13:32:21.105415838"
    },
    "analysis - 3 samples - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            
                        ]
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "analysis.ipynb:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    "versions.yml:md5,8a85556b3ab439938f8dc3b6f42b9b7f"
                ],
                "nichecompass_figures": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            
                        ]
                    ]
                ],
                "notebook": [
                    [
                        {
                            "id": "test"
                        },
                        "analysis.ipynb:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,8a85556b3ab439938f8dc3b6f42b9b7f"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-08-30T15:52:09.533313099"
    },
    "mouse - 3": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        [
                            [
                                [
                                    "batches_latent_physical_space.svg:md5,fd5e4586463e95a8fae2f33cb9a537a6",
                                    "cell_types_latent_physical_space.svg:md5,2e0e9605fbfb2d9a2596fb68be6e44a1",
                                    "enriched_gps_heatmap.svg:md5,329dbc45625d6befa3b1e743eac7760f",
                                    "log_bayes_factor_2.3_cluster_None_vs_rest_top_20-30_enriched_gps_gp_scores_omics_feature_counts_in_sagittal1_feature_space.png:md5,0bb3de9e244870b495c7a4e859068c95",
                                    "log_bayes_factor_2.3_cluster_None_vs_rest_top_20-30_enriched_gps_gp_scores_omics_feature_counts_in_sagittal2_feature_space.png:md5,278b20743d8bff14d5c61ee5b605dd5a",
                                    "log_bayes_factor_2.3_cluster_None_vs_rest_top_20-30_enriched_gps_gp_scores_omics_feature_counts_in_sagittal3_feature_space.png:md5,9522b0026c83fd301c2f1434084d4c14",
                                    "log_bayes_factor_2.3_niche_enriched_gps_summary.csv:md5,9c91cd8d1c09ef7d0f11b75b2b8cadb3",
                                    "res_0.2_niche_composition_batches.svg:md5,efd0c51a66a1c18830ab7ac666b7dc6c",
                                    "res_0.2_niche_composition_cell_types.svg:md5,27f2d845e6df890bdaf245a222509e5a",
                                    "res_0.2_niches_latent_physical_space.svg:md5,ad5f95274ba8ad0f36e42b3e85a1c740"
                                ],
                                [
                                    "attr.pkl:md5,0e9602560d2c665f3c383f9a850171a3",
                                    "model_params.pt:md5,c5e78c72684e62675cd1211e1f0fc046",
                                    "trained.h5ad:md5,d87b9fe09a0ee1611cde9f72b0c655c6",
                                    "var_names.csv:md5,d6db17b25f7a7847eac85d7b5a3552bc"
                                ]
                            ],
                            [
                                [
                                    "gencode.vM25.chr_patch_hapl_scaff.annotation.gtf.gz:md5,c77cbf3cad15af10b6174d4b77c8246f",
                                    "human_mouse_gene_orthologs.csv:md5,b01557bcc3647653d0b323e17fe1e93c"
                                ],
                                [
                                    [
                                        "human_metabolite_enzymes.tsv:md5,242269afd818d1ff93f190271065526c",
                                        "human_metabolite_sensors.tsv:md5,5a02dd9124c603d62e4944526f8b544c",
                                        "mouse_metabolite_enzymes.tsv:md5,f2a48c14af3f5174ae78cf390cbfbe0e",
                                        "mouse_metabolite_sensors.tsv:md5,1987eb7777861838ac8b6224732d88b8"
                                    ],
                                    "nichenet_ligand_target_matrix_v2_mouse.csv:md5,0c00fc0008112adaa551dbab0ce9a855",
                                    "nichenet_lr_network_v2_mouse.csv:md5,c42d99ee4976c592f477f58affff1cd1",
                                    "omnipath_lr_network.csv:md5,03487d78e4fe071fa42832be0bbdbf8d"
                                ]
                            ],
                            "run_config.json:md5,dbc58b55de2275ebfbd20f0a36b28386",
                            "timestamp.txt:md5,026972b602ac18dba8c7a3cf777e0bc7",
                            "train.log:md5,274339ccd3c8f5fe388209523e2e0cbf"
                        ]
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "analysis_20250828_170240.ipynb:md5,fa83dd72013a81d0ca6436c2e28a6084"
                    ]
                ],
                "2": [
                    "versions.yml:md5,8a85556b3ab439938f8dc3b6f42b9b7f"
                ],
                "nichecompass_dir": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        [
                            [
                                [
                                    "batches_latent_physical_space.svg:md5,fd5e4586463e95a8fae2f33cb9a537a6",
                                    "cell_types_latent_physical_space.svg:md5,2e0e9605fbfb2d9a2596fb68be6e44a1",
                                    "enriched_gps_heatmap.svg:md5,329dbc45625d6befa3b1e743eac7760f",
                                    "log_bayes_factor_2.3_cluster_None_vs_rest_top_20-30_enriched_gps_gp_scores_omics_feature_counts_in_sagittal1_feature_space.png:md5,0bb3de9e244870b495c7a4e859068c95",
                                    "log_bayes_factor_2.3_cluster_None_vs_rest_top_20-30_enriched_gps_gp_scores_omics_feature_counts_in_sagittal2_feature_space.png:md5,278b20743d8bff14d5c61ee5b605dd5a",
                                    "log_bayes_factor_2.3_cluster_None_vs_rest_top_20-30_enriched_gps_gp_scores_omics_feature_counts_in_sagittal3_feature_space.png:md5,9522b0026c83fd301c2f1434084d4c14",
                                    "log_bayes_factor_2.3_niche_enriched_gps_summary.csv:md5,9c91cd8d1c09ef7d0f11b75b2b8cadb3",
                                    "res_0.2_niche_composition_batches.svg:md5,efd0c51a66a1c18830ab7ac666b7dc6c",
                                    "res_0.2_niche_composition_cell_types.svg:md5,27f2d845e6df890bdaf245a222509e5a",
                                    "res_0.2_niches_latent_physical_space.svg:md5,ad5f95274ba8ad0f36e42b3e85a1c740"
                                ],
                                [
                                    "attr.pkl:md5,0e9602560d2c665f3c383f9a850171a3",
                                    "model_params.pt:md5,c5e78c72684e62675cd1211e1f0fc046",
                                    "trained.h5ad:md5,d87b9fe09a0ee1611cde9f72b0c655c6",
                                    "var_names.csv:md5,d6db17b25f7a7847eac85d7b5a3552bc"
                                ]
                            ],
                            [
                                [
                                    "gencode.vM25.chr_patch_hapl_scaff.annotation.gtf.gz:md5,c77cbf3cad15af10b6174d4b77c8246f",
                                    "human_mouse_gene_orthologs.csv:md5,b01557bcc3647653d0b323e17fe1e93c"
                                ],
                                [
                                    [
                                        "human_metabolite_enzymes.tsv:md5,242269afd818d1ff93f190271065526c",
                                        "human_metabolite_sensors.tsv:md5,5a02dd9124c603d62e4944526f8b544c",
                                        "mouse_metabolite_enzymes.tsv:md5,f2a48c14af3f5174ae78cf390cbfbe0e",
                                        "mouse_metabolite_sensors.tsv:md5,1987eb7777861838ac8b6224732d88b8"
                                    ],
                                    "nichenet_ligand_target_matrix_v2_mouse.csv:md5,0c00fc0008112adaa551dbab0ce9a855",
                                    "nichenet_lr_network_v2_mouse.csv:md5,c42d99ee4976c592f477f58affff1cd1",
                                    "omnipath_lr_network.csv:md5,03487d78e4fe071fa42832be0bbdbf8d"
                                ]
                            ],
                            "run_config.json:md5,dbc58b55de2275ebfbd20f0a36b28386",
                            "timestamp.txt:md5,026972b602ac18dba8c7a3cf777e0bc7",
                            "train.log:md5,274339ccd3c8f5fe388209523e2e0cbf"
                        ]
                    ]
                ],
                "notebook": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "analysis_20250828_170240.ipynb:md5,fa83dd72013a81d0ca6436c2e28a6084"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,8a85556b3ab439938f8dc3b6f42b9b7f"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-08-29T13:29:22.950591223"
    },
    "h5ad - sample_3 - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            [
                                [
                                    "attr.pkl:md5,0e9602560d2c665f3c383f9a850171a3",
                                    "model_params.pt:md5,c5e78c72684e62675cd1211e1f0fc046",
                                    "trained.h5ad:md5,d87b9fe09a0ee1611cde9f72b0c655c6",
                                    "var_names.csv:md5,d6db17b25f7a7847eac85d7b5a3552bc"
                                ]
                            ],
                            [
                                [
                                    "gencode.vM25.chr_patch_hapl_scaff.annotation.gtf.gz:md5,c77cbf3cad15af10b6174d4b77c8246f",
                                    "human_mouse_gene_orthologs.csv:md5,b01557bcc3647653d0b323e17fe1e93c"
                                ],
                                [
                                    [
                                        "human_metabolite_enzymes.tsv:md5,242269afd818d1ff93f190271065526c",
                                        "human_metabolite_sensors.tsv:md5,5a02dd9124c603d62e4944526f8b544c",
                                        "mouse_metabolite_enzymes.tsv:md5,f2a48c14af3f5174ae78cf390cbfbe0e",
                                        "mouse_metabolite_sensors.tsv:md5,1987eb7777861838ac8b6224732d88b8"
                                    ],
                                    "nichenet_ligand_target_matrix_v2_mouse.csv:md5,0c00fc0008112adaa551dbab0ce9a855",
                                    "nichenet_lr_network_v2_mouse.csv:md5,c42d99ee4976c592f477f58affff1cd1",
                                    "omnipath_lr_network.csv:md5,03487d78e4fe071fa42832be0bbdbf8d"
                                ]
                            ],
                            "run_config.json:md5,dbc58b55de2275ebfbd20f0a36b28386",
                            "timestamp.txt:md5,026972b602ac18dba8c7a3cf777e0bc7",
                            "train.log:md5,274339ccd3c8f5fe388209523e2e0cbf"
                        ]
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "analysis_20250828_170240.ipynb:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    "versions.yml:md5,8a85556b3ab439938f8dc3b6f42b9b7f"
                ],
                "nichecompass_dir": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            [
                                [
                                    "attr.pkl:md5,0e9602560d2c665f3c383f9a850171a3",
                                    "model_params.pt:md5,c5e78c72684e62675cd1211e1f0fc046",
                                    "trained.h5ad:md5,d87b9fe09a0ee1611cde9f72b0c655c6",
                                    "var_names.csv:md5,d6db17b25f7a7847eac85d7b5a3552bc"
                                ]
                            ],
                            [
                                [
                                    "gencode.vM25.chr_patch_hapl_scaff.annotation.gtf.gz:md5,c77cbf3cad15af10b6174d4b77c8246f",
                                    "human_mouse_gene_orthologs.csv:md5,b01557bcc3647653d0b323e17fe1e93c"
                                ],
                                [
                                    [
                                        "human_metabolite_enzymes.tsv:md5,242269afd818d1ff93f190271065526c",
                                        "human_metabolite_sensors.tsv:md5,5a02dd9124c603d62e4944526f8b544c",
                                        "mouse_metabolite_enzymes.tsv:md5,f2a48c14af3f5174ae78cf390cbfbe0e",
                                        "mouse_metabolite_sensors.tsv:md5,1987eb7777861838ac8b6224732d88b8"
                                    ],
                                    "nichenet_ligand_target_matrix_v2_mouse.csv:md5,0c00fc0008112adaa551dbab0ce9a855",
                                    "nichenet_lr_network_v2_mouse.csv:md5,c42d99ee4976c592f477f58affff1cd1",
                                    "omnipath_lr_network.csv:md5,03487d78e4fe071fa42832be0bbdbf8d"
                                ]
                            ],
                            "run_config.json:md5,dbc58b55de2275ebfbd20f0a36b28386",
                            "timestamp.txt:md5,026972b602ac18dba8c7a3cf777e0bc7",
                            "train.log:md5,274339ccd3c8f5fe388209523e2e0cbf"
                        ]
                    ]
                ],
                "notebook": [
                    [
                        {
                            "id": "test"
                        },
                        "analysis_20250828_170240.ipynb:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,8a85556b3ab439938f8dc3b6f42b9b7f"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-08-29T14:38:25.548303984"
    }
}